pkg/userdata/plugin/data.go

```diff
@@ -0,0 +1,43 @@
+//
+// Core UserData plugin.
+//
+
+package plugin
+
+import (
+	"net"
+
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+const (
+	// EnvRequest names the environment variable signalling the
+	// plugin which request the caller wants to have answered.
+	EnvRequest = "REQUEST"
```

Comment:
Can we put these into a package below `pkg/apis` so consuming doesn't result in importing the whole `userdataplugin` package?

Reply:
Actually thinking about this, is there even a reason why they need to be public?

Reply:
It's used inside the `plugin` package as well as in the `manager` package. Surely could declare it private and redundant as it is only interesting between these two packages to communicate.

Reply:
Thought about it. They also share the data types to serialise so API package might make sense. Otherwise I would expect by the linker to not keep unreferenced types of `userdata/plugin` in `machine-controller` binary.